src/colmap/util/threading.cc

```diff
@@ -194,6 +194,9 @@ ThreadPool::ThreadPool(const int num_threads)
 ThreadPool::~ThreadPool() {
   try {
```

Comment:
nit: I should have done this already but a short comment that the destructor shouldn't throw and that's why we do a fatal log.